api/main.py

```diff
@@ -17,10 +19,15 @@ async def upload(video: fastapi.UploadFile = fastapi.File(...)):
     await video.seek(0)
     content = await video.read()
 
-    print(content)
     print(video.filename)
     print(video.content_type)
-    open("/tmp/hello.mkv", "wb").write(content)
+
+    # save video
+    suffix = pathlib.Path(video.filename).suffix
+    file_hash = hashlib.md5(content).hexdigest()
+    filename = file_hash + suffix
```

Comment:
We could consider breaking out the filename generation to a separate function ;)